.hlint.yaml

```diff
@@ -28,6 +28,11 @@
   - {name: [Data.Set, Data.HashSet], as: Set}
   - {name: [Data.Map, Data.HashMap.Strict, Data.HashMap.Lazy], as: Map}
   - {name: Control.Arrow, within: []}
+  - {name: HypotheticalModule1, as: HM1, asRequired: true}
+  - {name: HypotheticalModule2, qualified: exhaustive}
```

Comment:
I'm not sure the meaning of `qualified: exhaustive` is immediately obvious. Perhaps `import_style: qualified / explicit / qualified_or_explicit`?

Reply:
:thinking: with the proposed `import_style` we lose the ability to just qualification without requiring an explicit import list. That said, I don't know if it's particularly useful.

Reply:
I've made the changes. Not sure if `unqualified` and `explicit` are particularly useful, but maybe there's a use case for those I didn't think of.

Reply:
> unqualified

I'd suggest removing this - "This module must be imported unqualified!" says no one, ever :smile:

Reply:
> says no one, ever

I would avoid cutting obvious functionality just because you can't think of a use case from the top of your head, it ultimately only leads to user frustration. It's not particularly costly to implement, and I'd like to have the option on the off-chance this comes in handy. One example where I imagine this might be somewhat useful is, say, custom prelude imports, which you might want to always be unqualified, but not necessarily always explicit.

Reply:
I actually quite like saying modules must be imported unqualified. Every time I see `import Data.List(sort)` I reply with just don't bother listing explicit imports. So I think it's possibly something I'd use 😄 .